Sources/KlaviyoSwift/AppContextInfo.swift

```diff
@@ -30,17 +48,76 @@ struct AppContextInfo {
         "\(osName) \(osVersion)"
     }
 
-    init(excutable: String = DEFAULT_EXECUTABLE,
+    init(executable: String = DEFAULT_EXECUTABLE,
          bundleId: String = DEFAULT_BUNDLE_ID,
          appVersion: String = DEFAULT_APP_VERSION,
          appBuild: String = DEFAULT_APP_BUILD,
+         appName: String = DEFAULT_APP_NAME,
          version: OperatingSystemVersion = DEFAULT_OS_VERSION,
-         osName: String = DEFAULT_OS_NAME) {
-        self.excutable = excutable
+         osName: String = DEFAULT_OS_NAME,
+         manufacturer: String = DEFAULT_MANUFACTURER,
+         deviceModel: String = DEFAULT_DEVICE_MODEL,
+         deviceId: String = UIDevice.current.identifierForVendor?.uuidString ?? "") {
+        self.executable = executable
         self.bundleId = bundleId
         self.appVersion = appVersion
         self.appBuild = appBuild
+        self.appName = appName
         self.version = version
         self.osName = osName
+        self.manufacturer = manufacturer
+        self.deviceModel = deviceModel
+        self.deviceId = deviceId
+
+        switch UIDevice.current.pushEnvironment {
+        case .development:
+            environment = "debug"
+        case .production:
+            environment = "release"
+        case .unknown:
+            #if DEBUG
+            environment = "debug"
+            #else
+            environment = "release"
+            #endif
+        }
+    }
+}
+
+extension UIDevice {
+    public enum PushEnvironment: String {
+        case unknown
+        case development
+        case production
+    }
+
+    public var pushEnvironment: PushEnvironment {
+        guard let provisioningProfile = try? provisioningProfile(),
+              let entitlements = provisioningProfile["Entitlements"] as? [String: Any],
+              let environment = entitlements["aps-environment"] as? String
+        else {
+            return .unknown
+        }
+
+        return PushEnvironment(rawValue: environment) ?? .unknown
+    }
+
+    // MARK: - Private
+
+    private func provisioningProfile() throws -> [String: Any]? {
+        guard let url = Bundle.main.url(forResource: "embedded", withExtension: "mobileprovision") else {
+            return nil
+        }
+
+        let binaryString = try String(contentsOf: url, encoding: .isoLatin1)
```

Comment:
do we want to optionally try this or is this try safe?

Reply:
yah +1!

Reply:
Oh wait there is an optional try above nvm.